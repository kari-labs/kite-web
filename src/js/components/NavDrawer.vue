<template>
    <v-navigation-drawer clipped app class="accent" v-model="drawer">
        <v-list dense class="pt-0">
            <v-list-tile v-for="item in items" :key="item.title" @click="handleRoute(item.route)">
                <v-list-tile-action>
                    <v-icon>{{ item.icon }}</v-icon>
                </v-list-tile-action>
                <v-list-tile-content>
                    <v-list-tile-title>{{ item.title }}</v-list-tile-title>
                </v-list-tile-content>
            </v-list-tile>
        </v-list>
    </v-navigation-drawer>
</template>

<script>
    export default {
        data: () => ({
            items: [
                {
                    title: 'Home',
                    icon: 'dashboard',
                    route: 'listContainers'
                },
                {
                    title: 'Report Bug',
                    icon: 'bug_report'
                }
            ],
            drawer: false
        }),
        methods: {
            handleRoute(routeName) {
                if(routeName) this.$router.push({ name: routeName})
            }
        }
    }
</script>