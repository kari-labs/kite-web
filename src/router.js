import Vue from 'vue'
import Router from 'vue-router'

Vue.use(Router)

export default new Router({
  mode: 'history',
  base: '/',
  routes: [
    {
        path: '/',
        name: 'listContainers',
        component: () => import(/* webpackChunkName: "containerDashboard" */ './views/ContainerDashboard.vue')
    }
  ]
})
