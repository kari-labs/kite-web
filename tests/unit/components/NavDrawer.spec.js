import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import router from '@/router'
import NavDrawer from '@/components/NavDrawer.vue'

describe('NavDrawer', () => {
    let wrapper

    beforeEach(() => {
        Vue.use(router)
        Vue.use(Vuetify)

        wrapper = mount(NavDrawer, { router })
    })

    it('should render Vuetify components', () => {
        expect(wrapper.contains('aside.v-navigation-drawer')).toBe(true)
    })

    it('should render drawer items correctly', () => {
        wrapper.vm.items = [
            {
                title: 'Foo',
                icon: 'dashboard',
                route: 'foo'
            },
            {
                title: 'Bar',
                icon: 'bug_report'
            },
            {
                title: 'FooBar',
                icon: 'fooBar',
                route: 'fooBar'
            }
        ]

        expect(wrapper.element).toMatchSnapshot()
    })

    it('handleRoute pushes correct named route', () => {
        wrapper.vm.handleRoute('listContainers')
        expect(wrapper.vm.$route.path).toBe('/' || '')
    })

    it('handleRoute should do NOTHING without params', () => {
        const initialPath = wrapper.vm.$route.path

        wrapper.vm.handleRoute()
        expect(wrapper.vm.$route.path).toBe(initialPath)
    })
})