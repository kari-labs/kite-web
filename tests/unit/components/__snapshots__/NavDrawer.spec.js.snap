// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer should render drawer items correctly 1`] = `
<aside
  class="accent v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light"
  style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(-300px); width: 300px;"
>
  <div
    class="v-list pt-0 v-list--dense theme--light"
  >
    <div>
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            dashboard
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Foo
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            bug_report
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Bar
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            fooBar
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            FooBar
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="v-navigation-drawer__border"
  />
</aside>
`;
